File: 132 Pattern.cpp
class Solution {
public:
    bool find132pattern(vector<int>& nums) {\
        int n=nums.size();
        if (n<=2)
         return false;
        // int i=0,j=2;
        // while(j<n){
        //     if (nums[i]<nums[j] && nums[j]<nums[i+1])
        //     return true;
        //     j++;
        //     i++;
        // }
        // return false;
         int currmin=INT_MIN;
        stack<int> s;
        for (int i=n-1;i>-1;i--)
        {
            if (nums[i]<currmin)
            return true;
            while (!s.empty() && nums[i]>s.top()){
            currmin=s.top();
            s.pop();
            }
           
            s.push(nums[i]);
        
        }
        return false;
    }
};

File: camp/week-2/leetcode/132-pattern.py
class Solution:
    def find132pattern(self, nums: List[int]) -> bool:
        if len(nums) <= 2:
            return False
        currMin = -inf
        stack = []
        for i in range(len(nums)-1,-1,-1):
            if nums[i] < currMin:
                return True
            while stack and stack[-1] < nums[i]:
                currMin = stack[-1]
                stack.pop()
            stack.append(nums[i])
        return False

